.github/workflows/deploy.yml

```diff
@@ -5,41 +5,66 @@ on:
     branches:
       - main
 
+# Set minimal permissions at workflow level
+permissions:
+  contents: read
+  deployments: write
+
 jobs:
   deploy:
     if: github.repository_owner == 'freifunkMUC'
+    # Job-specific permissions (inherits and extends workflow permissions)
+    permissions:
+      contents: read
+      deployments: write
```

Comment:
i am also unsure if we need deployments as the github-pages deploymend is not done via this file